plugins/gradle/src/org/jetbrains/plugins/gradle/util/GradleJvmResolutionUtil.kt

```diff
@@ -85,6 +90,7 @@ fun updateGradleJvm(project: Project, externalProjectPath: String) {
   val projectRootManager = ProjectRootManager.getInstance(project)
   val projectSdk = projectRootManager.projectSdk ?: return
   if (projectSdk.name != gradleJvm) return
+  if (GradleDaemonJvmHelper.isProjectUsingDaemonJvmCriteria(projectSettings)) return
```

Comment:
Why do you need this condition here? 

Here, we replace Gradle JVM with the Project SDK reference if they are the same. This assignment doesn't change the Gradle JVM criteria logic. So, we can keep this block straight without changes.

Reply:
Once project is using `Daemon Jvm criteria` the `Project SDK` will be updated after a successfully sync meaning that updating it again from different parts of the code shouldn't be allowed since we want to establish a unidirectional flow: 
```
 Daemon JVM criteria -> Gradle Daemon -> Project SDK
```